Automattic/wp-calypso
# Turn PluginsBrowser into a functional component

#### Changes proposed in this Pull Request

Turn PluginsBrowser into a functional component

#### Testing instructions
No visual change should be seen.

1. Go to `/plugins` page without a selected site.
1. Click on the breadcrumbs to see if it still works.
1. Search for a plugin and check if the list gets filtered.
1. Verify if clicking the **Manage plugins** button redirects you to `/plugins/manage`.
1. Verify if clicking the **Upload** button redirects you to `/plugins/upload`.
1. Select a site and redo the steps from **2** to **5**.

---

Fixes #59002
Also related to #58453 by removing all lifecycle methods.
